driver-reactive-streams/src/test/functional/com/mongodb/reactivestreams/client/ClientEncryptionCustomEndpointTest.java

```diff
@@ -193,7 +193,7 @@ public void testCustomEndpoint() throws Throwable {
                         + "  key: \"arn:aws:kms:us-east-1:579766882180:key/89fcc2c4-08b0-4bd9-9f25-e30687b580d0\",\n"
                         + "  endpoint: \"kms.us-east-1.amazonaws.com:12345\"\n"
                         + "}"),
-                false, MongoClientException.class, ConnectException.class, "Connection refused"});
+                false, MongoClientException.class, MongoSocketOpenException.class, "Exception opening socket"});
```

Comment:
How com this had to change?

Reply:
I had missed the unwrapping of the exceptions in the KeyManagementService.

Reply:
Fixed